YearInReview.UnitTests/Model/Aggregators/AddedGamesAggregatorTest.cs

```diff
@@ -0,0 +1,256 @@
+			var allDatesOfYear = GetAllDatesOfYear(year);
+			// create games with added date for each day of the year
+			var games = allDatesOfYear.Select(date => new Game { Added = date }).ToList();
+			
+			// Act
+			var result = sut.GetAddedGames(games, year);
+			
+			// Assert
+			Assert.Equal(games.Count, result.Count);
+		}
+
+		[Theory] 
+		[AutoFakeItEasyData] 
+		public void GetAddedGames_ReturnsAllCorrect_WhenAllGamesMatching_AllDaysInLeapYear(AddedGamesAggregator sut)
+		{
+			// Arrange
+			const int year = 2024;
+			// all dates of a full leap year in one list
+			var allDatesOfYear = GetAllDatesOfYear(year);
+			// create games with added date for each day of the year
+			var games = allDatesOfYear.Select(date => new Game { Added = date }).ToList();
+			
+			// Act
+			var result = sut.GetAddedGames(games, year);
+			
+			// Assert
+			Assert.Equal(games.Count, result.Count);
+		}
+		
+		[Theory]
+		[AutoFakeItEasyData]
+		public void GetAddedGames_ReturnsCorrectGames_WhenMultipleGamesWithSameAddedDate(AddedGamesAggregator sut)
+		{
+			// Arrange
+			var date = new DateTime(2025, 1, 1);
+			var game1 = new Game { Added = date };
+			var game2 = new Game { Added = date };
+			var game3 = new Game { Added = date };
+
+			var games = new List<Game> { game1, game2, game3 };
+
+			// Act
+			var result = sut.GetAddedGames(games, date.Year);
+
+			// Assert
+			Assert.Equal(3, result.Count);
+			Assert.Contains(result, r => Equals(r.Game, game1) && r.AddedDate == date);
+			Assert.Contains(result, r => Equals(r.Game, game2) && r.AddedDate == date);
+			Assert.Contains(result, r => Equals(r.Game, game3) && r.AddedDate == date);
+		}
+
+		[Theory]
+		[AutoFakeItEasyData]
+		public void GetAddedGames_ReturnsOnlyMatching_WhenIncludesGamesAddedOnBoundaryDates(AddedGamesAggregator sut)
+		{
+			// Arrange
+			const int year = 2025;
+			var firstDayOfYear = new DateTime(year, 1, 1);
+			var lastDayOfYear = new DateTime(year, 12, 31);
+			var gameOnFirstDay = new Game { Added = firstDayOfYear };
+			var gameOnLastDay = new Game { Added = lastDayOfYear };
+			var gameOutsideYear1 = new Game { Added = new DateTime(year + 1, 1, 1) };
+			var gameOutsideYear2 = new Game { Added = new DateTime(year - 1, 12, 31) };
+
+			var games = new List<Game> { gameOnFirstDay, gameOnLastDay, gameOutsideYear1, gameOutsideYear2 };
+
+			// Act
+			var result = sut.GetAddedGames(games, year);
+
+			// Assert
+			Assert.Equal(2, result.Count);
+			Assert.Contains(result, r => Equals(r.Game, gameOnFirstDay) && r.AddedDate == firstDayOfYear);
+			Assert.Contains(result, r => Equals(r.Game, gameOnLastDay) && r.AddedDate == lastDayOfYear);
+			Assert.DoesNotContain(result, r => Equals(r.Game, gameOutsideYear1));
+			Assert.DoesNotContain(result, r => Equals(r.Game, gameOutsideYear2));
+		}
+		
+		[Theory]
+		[AutoFakeItEasyData]
+		public void GetAddedGames_ReturnsCorrectGames_WhenIncludesDifferentNullValues(AddedGamesAggregator sut)
```

Comment:
I think either in Roy Osherove's "The Art of Unit Testing" or in another authors "Working with Legacy Code" one of the characteristics of a good unit test was described as: it's precise and helps to pinpoint an issue when it fails.

Since this test is testing a combined rule set, if this test was failing it would not be clear what went wrong. While you might want to have a more "production" like inputs, it could also introduce brittleness (tests break easier, are harder to maintain) and again, it does not pinpoint specific issue.

Food for thought ;)